Update features and feature dependencies

NikVolf opened the pull request.

NikVolf commented:
I also moved SmallVec and fixed arrays implementations from ethereum feature (haven't realised they were there 🥺

ordian commented:
> I also moved SmallVec and fixed arrays implementations from ethereum feature (haven't realised they were there 🥺

Good catch! Might make sense to rename `impl.rs` to `ethereum_impls.rs` as well.

NikVolf commented:
@dvdplm I moved smallvec tests and updated docs

having dedicated module for ethereum impls (now `ethereum_impls`) seems fine to me, even if it just has couple of macro invocations
